### DISCUSSION TITLE
Deprecation Warnings

### DISCUSSION THREAD

**Discussion by schengal4** [safevideo/autollm#237] [Q&A] [answered]
**I ran the following code successfully**:

from autollm import AutoQueryEngine, read_files_as_documents

documents = read_files_as_documents(input_dir="data")
query_engine = AutoQueryEngine.from_defaults(documents)

response = query_engine.query("Summarize this essay.")
print(response)

**However, I'm getting the following warnings (mostly deprecation warnings)**:

`LangChainDeprecationWarning: Importing chat models from langchain is deprecated. Importing from langchain will no longer be supported as of langchain==0.2.0. Please import from langchain-community instead:

`from langchain_community.chat_models import ChatAnyscale`.

To install langchain-community run `pip install -U langchain-community`.
  warnings.warn(
C:\Users\vpera\OneDrive\Documents\GitHub\OncologyClinicalTrial\.env\lib\site-packages\langchain\chat_models\__init__.py:31: LangChainDeprecationWarning: Importing chat models from langchain is deprecated. Importing from langchain will no longer be supported as of langchain==0.2.0. Please import from langchain-community instead:

`from langchain_community.chat_models import ChatOpenAI`.

To install langchain-community run `pip install -U langchain-community`.
  warnings.warn(
C:\Users\vpera\OneDrive\Documents\GitHub\OncologyClinicalTrial\.env\lib\site-packages\langchain\embeddings\__init__.py:29: LangChainDeprecationWarning: Importing embeddings from langchain is deprecated. Importing from langchain will no longer be supported as of langchain==0.2.0. Please import from langchain-community instead:

`from langchain_community.embeddings import HuggingFaceBgeEmbeddings`.

To install langchain-community run `pip install -U langchain-community`.
  warnings.warn(
C:\Users\vpera\OneDrive\Documents\GitHub\OncologyClinicalTrial\.env\lib\site-packages\langchain\embeddings\__init__.py:29: LangChainDeprecationWarning: Importing embeddings from langchain is deprecated. Importing from langchain will no longer be supported as of langchain==0.2.0. Please import from langchain-community instead:

`from langchain_community.embeddings import HuggingFaceEmbeddings`.

To install langchain-community run `pip install -U langchain-community`.
  warnings.warn(
C:\Users\vpera\OneDrive\Documents\GitHub\OncologyClinicalTrial\.env\lib\site-packages\langchain\llms\__init__.py:548: LangChainDeprecationWarning: Importing LLMs from langchain is deprecated. Importing from langchain will no longer be supported as of langchain==0.2.0. Please import from langchain-community instead:        

`from langchain_community.llms import AI21`.

To install langchain-community run `pip install -U langchain-community`.
  warnings.warn(
C:\Users\vpera\OneDrive\Documents\GitHub\OncologyClinicalTrial\.env\lib\site-packages\langchain\llms\__init__.py:548: LangChainDeprecationWarning: Importing LLMs from langchain is deprecated. Importing from langchain will no longer be supported as of langchain==0.2.0. Please import from langchain-community instead:        

`from langchain_community.llms import Cohere`.

To install langchain-community run `pip install -U langchain-community`.
  warnings.warn(
C:\Users\vpera\OneDrive\Documents\GitHub\OncologyClinicalTrial\.env\lib\site-packages\langchain\llms\__init__.py:548: LangChainDeprecationWarning: Importing LLMs from langchain is deprecated. Importing from langchain will no longer be supported as of langchain==0.2.0. Please import from langchain-community instead:        

`from langchain_community.llms import FakeListLLM`.

To install langchain-community run `pip install -U langchain-community`.
  warnings.warn(
C:\Users\vpera\OneDrive\Documents\GitHub\OncologyClinicalTrial\.env\lib\site-packages\langchain\llms\__init__.py:548: LangChainDeprecationWarning: Importing LLMs from langchain is deprecated. Importing from langchain will no longer be supported as of langchain==0.2.0. Please import from langchain-community instead:        

`from langchain_community.llms import OpenAI`.

To install langchain-community run `pip install -U langchain-community`.
  warnings.warn(
2024-02-14 11:13:45,111 - autollm - WARNING - use_async is only supported for SimpleVectorStore. Ignoring use_async.`

**Comment by talhasarit** [comment, marked as answer]
Hi @schengal4,

Thanks for your message. The deprecation warnings you're seeing are due to older import paths from `langchain`. However, `autollm` primarily uses `llama-index` for its operations, and these warnings do not impact its core functionality. 

You can safely continue using `autollm` as these warnings are more about future compatibility and don't affect the current operations. We'll ensure to update these dependencies in future releases to align with `langchain-community` standards.
